R/BackendHdf5-class.R

```diff
@@ -257,11 +257,11 @@ setMethod("backendReadSpectra", "BackendHdf5", function(object, spectraData,
     h5 <- H5Fopen(h5file)
     on.exit(invisible(H5Fclose(h5)))
     comp_level <- .hdf5_compression_level()
-    if (prune) {
-        h5delete(h5, "spectra")
-    }
-    if (!H5Lexists(h5, "spectra"))
-        h5createGroup(h5, "spectra")
+    x <- force(x)
```

Comment:
Do we really need `force` here? To be honest I never used it before.

Reply:
Yes! I had it without and got very weird errors. In fact it's R's lazy evaluation. I passed `x = spectra(object)` to the function and R was first deleting the file contents and then evaluating the `spectra(object)` expression - but the files were already empty. So, with `force` we make sure that we first read the spectra from the files **before** we delete their content.

Reply:
Ok, makes sense, thanks.